base/operators.jl

```diff
@@ -293,7 +293,7 @@ fldmod1{T<:Real}(x::T, y::T) = (fld1(x,y), mod1(x,y))
 fldmod1{T<:Integer}(x::T, y::T) = (fld1(x,y), mod1(x,y))
 
 # transpose
-transpose(x) = x
+transpose(x) = error("transpose not implemented for $(typeof(x)). Consider permutedims.")
```

Comment:
Why is this an `error` and not something like `ArgumentError`? I thought we were trying to move away from `error()`?